transport-native-epoll/src/main/java/io/netty/channel/epoll/AcceptedSocket.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.epoll;
+
+final class AcceptedSocket {
```

Comment:
we could just create big-enough byte array that can hold ipv4 / ipv6 / ipv4-mapped addresses but then we still would need a way to know how much of the data in the array we really need to read. I guess we could also just add a marker in the array but I don't think this is any better.